tinygrad/codegen/linearizer.py

```diff
@@ -252,84 +272,35 @@ def end_loop(xx:List[Variable]):
     self.load_cache: Dict[str, UOp] = {}
 
     # reduce op
-    fake_reduce_idxs = []
+    fake_reduce_idxs: List[Variable] = []
+    acc_idxs = global_idxs+local_idxs+fake_reduce_idxs+upcast_idxs
     if self.reduceop is not None:
       # define indexes
       reduce_idxs = [Variable(f"ridx{i}", 0, self.full_shape[i]-1) for i in range(self.first_reduce+len(self.group_for_reduce), self.shape_len-self.upcasted)]
       fake_reduce_idxs = [x*0 for x in reduce_idxs]
+      acc_idxs = global_idxs+local_idxs+fake_reduce_idxs+(aliased_wmma_idxs[2] if self.tensor_core else [])+upcast_idxs
 
       # define accumulator
-      acc = self.global_load(0, global_idxs+local_idxs+fake_reduce_idxs+upcast_idxs, {ReduceOps.SUM: 0.0, ReduceOps.MAX: -math.inf}[cast(ReduceOps, self.reduceop.op)])
+      acc = self.global_load(0, acc_idxs, {ReduceOps.SUM: 0.0, ReduceOps.MAX: -math.inf}[cast(ReduceOps, self.reduceop.op)])
 
       # reduce loop
       render_loop(reduce_idxs)
 
-      # barrier for fast GEMM
-      if self.use_tensor_cores: self.uop(UOps.BARRIER, None, ())
-
-      # compute local aliases
-      # TODO: this is garbage code and should be at least moved elsewhere
-      locals_to_store = []
-      for i in self.local_alias:
```

Comment:
What replaces this? A key aspect of the old implementation is that it would work with locals as well as tensor cores